File: Python/dsaOne/Blind 75/Javascript/Arrays_and_Hashing/longestConsecutiveSequence.js
//https://leetcode.com/problems/longest-consecutive-sequence/description/

const longestConsecutive = (nums) => {
    let set = new Set(nums)
    let longVal = 0

    for(n of set) {
        let lcVal = 0
        if(!set.has(n-1)) {
            while(set.has(n + lcVal)) {
                lcVal += 1
            }
            longVal = Math.max(lcVal, longVal)
        }
    }

    return longVal
}

console.log(longestConsecutive([100,4,200,1,3,2]))
console.log(longestConsecutive([0,3,7,2,5,8,4,6,0,1]))

File: Python/dsaOne/Blind 75/Arrays_and_Hashing/longestConsecutiveSequence.py
#https://leetcode.com/problems/longest-consecutive-sequence/

def longestConsecutiveSequence(nums):
    ns = set(nums)
    mx = 0

    for i in ns:
        l = 0
        if i-1 not in ns:
            while(i+l in ns):
                l += 1
            mx = max(mx, l)

    return mx
        
print(longestConsecutiveSequence([100,4,200,1,3,2]))
print(longestConsecutiveSequence([0,3,7,2,5,8,4,6,0,1]))